src/dataset_processor.py

```diff
@@ -205,10 +205,43 @@ def load_assigned_reads(save_file_name, gffutils_db, multimapped_chr_dict):
         yield current_gene_info, read_storage
 
 
+def construct_models_in_parallel(sample, chr_id, dump_filename, args, multimapped_reads, current_chr_record,
+                                 io_support):
+    processor = DatasetProcessor(args, parallel=True)
+    processor.create_aggregators(sample)
+    gffutils_db = gffutils.FeatureDB(args.genedb, keep_order=True)
+    logger.info("Processing chromosome " + chr_id)
+    transcripts = []
+    tmp_gff_printer = GFFPrinter(sample.out_dir, sample.label, io_support)
```

Comment:
I still feel like models should be stored in memory, and later renamed and dumped

Reply:
maybe later. but feel free to implement it :)

Reply:
nice try